ONSdigital/ras-collection-instrument
# CI Service - needs functionality to be able to upload a non-BRES SEFT Survey.

**Scope**
We need to be able to upload non BRES SEFT surveys, and to achieve this we must be able to
1. Upload with RU
2. Upload without RU
3. Upload with dynamic classifiers
4. Search with dynamic classifiers
5. Limit the search results in date order (most recent first)

**How to test**
The best way to test, is run the application (pipenv run python run.py) If you do this, you will need to update the config file so it points to a valid survey and exercise services. Additionally you will need to set a ONS_CRYPTOKEY (export ONS_CRYPTOKEY='anything')

Once up and running you should be able to **POST** to the end points with a **file**.

1. Upload with RU
http://0.0.0.0:8002/collection-instrument-api/1.0.2/upload/14fb3e68-4dca-46db-bf49-04b84e07e77c/999
2. Upload without RU
http://0.0.0.0:8002/collection-instrument-api/1.0.2/upload/14fb3e68-4dca-46db-bf49-04b84e07e77c
3. Upload with dynamic classifiers
http://0.0.0.0:8002/collection-instrument-api/1.0.2/upload/14fb3e68-4dca-46db-bf49-04b84e07e77c/999?classifiers={"FORM_TYPE": "001"}

And **GET**

4. Search with dynamic classifiers
http://0.0.0.0:8002/collection-instrument-api/1.0.2/collectioninstrument?searchString={"FORM_TYPE":%20"001"}
5. Limit the search results in date order (most recent first)
http://0.0.0.0:8002/collection-instrument-api/1.0.2/collectioninstrument?limit=2

N.B searchString can be multiple and can be used with limit i.e
http://0.0.0.0:8002/collection-instrument-api/1.0.2/collectioninstrument?searchString={"FORM_TYPE":%20"001", "RU":%20"999"}&limit=1
